src/api/github/push.ts

```diff
@@ -0,0 +1,198 @@
+import fs from 'fs/promises';
+
+import { graphql } from '@octokit/graphql';
+import type {
+  CreateCommitOnBranchInput,
+  FileAddition,
+  FileDeletion,
+} from '@octokit/graphql-schema';
+
+import * as Git from '../git';
+import type { ChangedFile } from '../git/getChangedFiles';
+
+import { apiTokenFromEnvironment } from './environment';
+
+interface CreateCommitResult {
+  createCommitOnBranch: {
+    commit: {
+      id: string;
+    };
+  };
+}
+
+interface CommitAndPushAllChangesParams {
+  dir: string;
+  /**
+   * The branch name
+   */
+  branch: string;
+  /**
+   * The headline of the commit message
+   */
+  messageHeadline: string;
+  /**
+   * The body of the commit message
+   */
+  messageBody?: string;
+  /**
+   * Updates the local git working directory to reflect the new remote state
+   */
+  updateLocal?: boolean;
+}
+
+export const commitAndPushAllChanges = async ({
+  dir,
+  branch,
+  messageHeadline,
+  messageBody,
+  updateLocal,
+}: CommitAndPushAllChangesParams): Promise<string> => {
+  const changedFiles = await Git.getChangedFiles({ dir });
+  const fileChanges = await mapChangedFilesToFileChanges(changedFiles);
+
+  return await commitAndPush({
+    dir,
+    branch,
+    messageHeadline,
+    messageBody,
+    fileChanges,
+    updateLocal,
+  });
+};
+
+interface FileChanges {
+  additions: FileAddition[];
+  deletions: FileDeletion[];
```

Comment:
I think `@octokit/graphql-schema` may need to be under `dependencies` because this type is part of our public interface